vowpalwabbit/learner.h

```diff
@@ -465,101 +550,110 @@ VW_WARNING_STATE_POP
     ret.learn_fd.multipredict_f = nullptr;
     ret.pred_type = pred_type;
     ret.is_multiline = std::is_same<multi_ex, E>::value;
+    ret.predict_before_learn = predict_before_learn;
+
+    VW_DBG_0 << "Added Reduction: " << name << std::endl;
 
     return ret;
   }
 };
 
 template <class T, class E, class L>
 learner<T, E>& init_learner(free_ptr<T>& dat, L* base, void (*learn)(T&, L&, E&), void (*predict)(T&, L&, E&),
-    size_t ws, prediction_type_t pred_type)
+    size_t ws, prediction_type_t pred_type, const std::string& name, bool predict_before_learn = true)
 {
-  auto ret = &learner<T, E>::init_learner(dat.get(), base, learn, predict, ws, pred_type);
+  auto ret = &learner<T, E>::init_learner(dat.get(), base, learn, predict, ws, pred_type, name, predict_before_learn);
```

Comment:
what exactly do you mean with "predict_before_learn" when it is true vs when it is false? is this state necessary?

Reply:
I read the code, but I'm curious about the intent of keeping track of this state.

Reply:
I covered this in another comment.